packages/server/utils/atlassian/jiraImages.ts

```diff
@@ -48,40 +44,42 @@ export const downloadAndCacheImages = async (
   manager: AtlassianManager,
   imageUrlToHash: Record<string, string>
 ) => {
-  await Promise.all(
+  return Promise.all(
     Object.entries(imageUrlToHash).map(([imageUrl, hash]) =>
       downloadAndCacheImage(manager, hash, imageUrl)
     )
   )
 }
 
-const createParabolImageUrl = (hashedImageUrl: string) => {
-  return `/jira-attachments/${hashedImageUrl}`
-}
-
-const downloadAndCacheImage = async (
+export const downloadAndCacheImage = async (
   manager: AtlassianManager,
-  hashedImageUrl: string,
+  imageUrlHash: string,
   imageUrl: string
 ) => {
+  const imageKey = `jira-image:${imageUrlHash}`
+  const imageMimeTypeKey = `jira-image:mime-type:${imageUrlHash}`
+
   const redis = getRedis()
-  const isImageAlreadyCached = await redis.exists(hashedImageUrl)
+  const isImageAlreadyCached = await redis.exists(imageKey)
   if (isImageAlreadyCached) return
 
-  await redis.setBuffer(hashedImageUrl, NO_IMAGE_BUFFER, 'PX', IMAGE_TTL_MS)
-  const imageBuffer = await manager.getImage(imageUrl)
-  if (!imageBuffer) {
-    await redis.del(hashedImageUrl)
+  await redis.setBuffer(imageKey, NO_IMAGE_BUFFER, 'PX', IMAGE_TTL_MS)
+  const imageResponse = await manager.getImage(imageUrl)
+  if (!imageResponse || !imageResponse.contentType) {
+    await redis.del(imageKey)
     return
   }
-  return redis.setBuffer(hashedImageUrl, imageBuffer, 'PX', IMAGE_TTL_MS)
+
+  return Promise.all([
+    redis.setBuffer(imageKey, imageResponse.imageBuffer, 'PX', IMAGE_TTL_MS),
+    redis.set(imageMimeTypeKey, imageResponse.contentType, 'PX', IMAGE_TTL_MS)
```

Comment:
+1 why not use `redis.multi` here?

Reply:
`mset` does not support setting expiration time, that's why I used it that way. If there's any better way, I'm happy to improve

Reply:
Wouldn't
```
return redis.multi()
    .setBuffer(imageKey, imageResponse.imageBuffer, 'PX', IMAGE_TTL_MS)
    .set(imageMimeTypeKey, imageResponse.contentType, 'PX', IMAGE_TTL_MS)
    .exec()
```
work just fine?

Reply:
you're right. not sure why, but I mistook `multi` with the `mset`, [which can't handle setting expiration times.](https://stackoverflow.com/questions/63722979/does-ioredis-mset-accepts-options-to-set-expiry-like-set-operation-does). I fixed it in both places, thank you!